### PR TITLE
[5.7] Check the module name to know when to skip inherited doc comments.

### PR DESCRIPTION
**Pull request by d-ronnqvist**
- **Rationale:** This is—together with the two other parts—fix an issue where documentation comments that are inherited from other modules may raise non-actionable warnings if the original documentation comment linked to other symbols in that module.
- **Risk:** Low
- **Risk Detail:** This uses the new API from https://github.com/apple/swift-docc-symbolkit/pull/43 to more reliably determine if inherited documentation comments originate from another module or from the current module.
- **Reward:** Medium
- **Reward Details:** Documentation builds successfully but documentation for protocol conformance symbols from other modules may render with broken links to symbols.
- **Original PR:** #314 
- **Issue:** rdar://92185538
- **Code Reviewed By:** @QuietMisdreavus 
- **Testing Details:** See https://github.com/apple/swift-docc/pull/314

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -124,7 +124,7 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(name: "swift-markdown", url: "https://github.com/apple/swift-markdown.git", .branch("release/5.7")),
         .package(name: "CLMDB", url: "https://github.com/apple/swift-lmdb.git", .branch("release/5.7")),
         .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "1.0.1")),
-        .package(name: "SymbolKit", url: "https://github.com/apple/swift-docc-symbolkit", .branch("release/5.7")),
+        .package(name: "SymbolKit", url: "https://github.com/d-ronnqvist/swift-docc-symbolkit", .branch("check-doc-comment-module-info-5.7")),
```
Comment (franklinsch) [inline Package.swift:127]: Please don't forget to undo this before merging. Thanks!

**Review by franklinsch — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Package.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Package.swift:127]: I just need to wait for https://github.com/apple/swift-docc-symbolkit/pull/43 to be merged before undoing this change.

Comment (d-ronnqvist) [reply Package.swift:127]: I'm holding this until I can cherry-pick https://github.com/apple/swift-docc-symbolkit/pull/44 as well

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Package.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Package.swift]: This is addressed in https://github.com/apple/swift-docc/pull/316/commits/094931d75128207ed6cf720b42f16c26b9106d13

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**